src/fastfield/reader.rs

```diff
@@ -61,6 +62,34 @@ pub trait FastFieldReader<Item: FastValue>: Clone {
     fn max_value(&self) -> Item;
 }
 
+struct FFHeader {
+    field_id: u8,
+    gcd: u64,
+    min_value: u64,
+}
+
+fn read_header(bytes: &mut OwnedBytes) -> FFHeader {
+    let magic_number_or_field_id = bytes.read_u8();
+    if magic_number_or_field_id == FF_HEADER_MAGIC_NUMBER {
+        let _header_version = bytes.read_u8();
+        let field_id = bytes.read_u8();
+        let gcd = bytes.read_u64();
+        let min_value = bytes.read_u64();
+        FFHeader {
+            field_id,
+            gcd,
+            min_value,
+        }
+    } else {
+        // old version
+        FFHeader {
+            field_id: magic_number_or_field_id,
```

Comment:
wait so today we cannot have more than 256 fields!?

Reply:
terrible naming, it should be `magic_number_or_codec_id`, I think we use u16 for field ids